src/services/blockchain/clients/midnight/client.rs

```diff
@@ -210,30 +267,49 @@ impl<T: Send + Sync + Clone + BlockchainTransport> BlockChainClient for Midnight
 	) -> Result<Vec<BlockType>, anyhow::Error> {
 		let block_futures: Vec<_> = (start_block..=end_block.unwrap_or(start_block))
 			.map(|block_number| {
-				let params = json!([
-					format!("0x{:x}", block_number),
-					true // include full transaction objects
-				]);
+				let params = json!([format!("0x{:x}", block_number)]);
 				let client = self.http_client.clone();
 
 				async move {
 					let response = client
-						.send_raw_request("eth_getBlockByNumber", Some(params))
+						.send_raw_request("chain_getBlockHash", Some(params))
+						.await
+						.with_context(|| {
+							format!("Failed to get block hash for: {}", block_number)
+						})?;
+
+					let block_hash = response
+						.get("result")
+						.and_then(|v| v.as_str())
+						.ok_or_else(|| anyhow::anyhow!("Missing 'result' field"))?;
+
+					let params = json!([block_hash]);
+
+					let response = client
+						.send_raw_request("midnight_jsonBlock", Some(params))
```

Comment:
Its interesting that all the prefixes in RPC methods are different, `mindinght_` `chain_` `system_`

Reply:
Some are specific to polkadot and some are specific to midnight, we need both :')